### PR TITLE
Create CF script to create IAM role with secret manager access

### PR DESCRIPTION
**Pull request by alex-addweb**

### CODE REVIEW

File: aws-secret-manager_with_-iam-role.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+"AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "",
+    "Parameters": {
+    },
+    "Resources": {
+        "SMIamRole" : {
+            "Type" : "AWS::IAM::Role",
+            "Properties" : {
+                "RoleName" : "Secret-Manager-Role",
```
Comment (jpoolGW) [inline aws-secret-manager_with_-iam-role.json:10]: Update role name to AWS-TestEngine-IAMRole

File: aws-secret-manager_with_-iam-role.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                            "Version": "2012-10-17",
+                            "Statement": [
+                                {
+                                    "Effect": "Allow",
+                                    "Action": [
+                                        "secretsmanager:*"
+                                    ],
+                                    "Resource": {"Ref" : "SecretManager"}
+                                }
+                            ]
```
Comment (jpoolGW) [inline aws-secret-manager_with_-iam-role.json:40]: Please update policy to access the specific secret. I will provide the secret ARN on slack when requested
Please also add a policy for S3 access. I will also provide ARN for the bucket

**Review by jpoolGW — CHANGES_REQUESTED** [review]

**Comment by alex-addweb** [comment]
@jpoolGW , S3 permission added and secretsmanager permission has been changed. Kindly chec and confirm

File: jmeter-icap-poc/cloudformation/aws-secret-manager_with_-iam-role.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                      "Value": "App"
+                    }
+                  ]
+                }
+              }
+        },
+        "Outputs": {
+    
+        }
+    }
```
Comment (jpoolGW) [inline jmeter-icap-poc/cloudformation/aws-secret-manager_with_-iam-role.json:116]: This is not the correct implementation.

We need a single role with IAM Policies for the required permissions.

**Review by jpoolGW — CHANGES_REQUESTED** [review]

File: jmeter-icap-poc/cloudformation/aws-secret-manager_with_-iam-role.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                "RoleName" : "AWS-TestEngine-IAMRole",
+                "AssumeRolePolicyDocument" : {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {
+                                "Service": [
+                                    "secretsmanager.amazonaws.com",
+                                    "s3.amazonaws.com"
```
Comment (jpoolGW) [inline jmeter-icap-poc/cloudformation/aws-secret-manager_with_-iam-role.json:23]: I think this is to allow services access to this role. The role will be accessed by EC2

**Review by jpoolGW — CHANGES_REQUESTED** [review]

**Review by jpoolGW — APPROVED** [review]

**Pull request merged**